scripts/install-cni.sh

```diff
@@ -110,15 +115,20 @@ else
   cni_spec=${cni_spec//@cniBandwidthPlugin/}
 fi
 
-token=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
 host=${KUBERNETES_SERVICE_HOST}
 # If host contains a colon (:), it is an IPv6 address, hence needs wrapping
 # with [..].
 if [[ "${host}" =~ : ]]; then
   host="[$host]"
 fi
-node_url="https://$host:${KUBERNETES_SERVICE_PORT}/api/v1/nodes/${HOSTNAME}"
-response=$(curl -k -s -H "Authorization: Bearer $token" "$node_url")
+node_url="https://$host:${KUBERNETES_SERVICE_PORT}/api/v1/watch/nodes/${HOSTNAME}"
+for _ in {1..3}; do
+  response=$(grep -m1 . <(curl -k -s -N -H "Authorization: Bearer $(</var/run/secrets/kubernetes.io/serviceaccount/token)" "$node_url" | jq --unbuffered -c '.object | select(.spec.podCIDR != null)'))
```

Comment:
there is no timeout specified, please specify the timeout explicitly using the `timeoutSeconds` request param to avoid  the forever hanging call.

Reply:
Additionally you can add the `limit=1` parameter to end the call as soon as the first update on the node object gets returned.

Reply:
Timeout added, and no, we can't end as soon as the first update, in case the first update is not adding podCIDR.

`grep -m1 . <(...)` can return after receiving the first line.